tests/test_diffusion.py

```diff
@@ -115,6 +116,46 @@ def single_step(u, ui):
     return u if ti % 2 == 0 else ui
 
 
+def execute_devito(ui, dx=0.01, dy=0.01, a=0.5, timesteps=500):
+    """Execute diffusion stencil using the devito Operator API."""
+    nx, ny = int(1 / dx), int(1 / dy)
+    dx2, dy2 = dx**2, dy**2
+    dt = dx2 * dy2 / (2 * a * (dx2 + dy2))
+    # Allocate the grid and set initial condition
+    # Note: This should be made simpler through the use of defaults
+    u = TimeData(name='u', spc_shape=(nx, ny), time_dim=-666, time_order=1,
+                 save=False, dtype=np.float32)
+    u.data[0, :] = ui[:]
+
+    # Derive the stencil according to devito conventions
+    def diffusion_stencil_devito():
+        # Note: Apparently 2D grid indexing is assumed to be (x, z)
```

Comment:
Yes, that's the plan. Preliminary version of this is in `automate-symbolic-substitution`.